src/ESPEasy_checks.ino

```diff
@@ -58,7 +58,12 @@ void run_compiletime_checks() {
   check_size<rulesTimerStatus,                      12u>();
   check_size<portStatusStruct,                      4u>();
   check_size<ResetFactoryDefaultPreference_struct,  4u>();
-  check_size<GpioFactorySettingsStruct,             17u>();
+#ifdef HAS_ETHERNET
+  const unsigned int GpioFactorySettingsStructSize = 17;
+#else
+  const unsigned int GpioFactorySettingsStructSize = 11;
```

Comment:
Not sure if this is practical.
If switching between builds with and without ethernet, then the settings file will not be exchangable.
The same for the SettingsStruct.
Just leave the member variables in the settings file, or else we will run into issues later when new members are added.

Reply:
Ok :)